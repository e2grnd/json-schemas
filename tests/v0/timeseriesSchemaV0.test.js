const fs = require('fs');
const path = require('path');
const Ajv = require('ajv/dist/2020').default;
const ajv = new Ajv({ strict: false, allErrors: true});

const timeseriesSchema = require('../../v0/input.timeseries.schema.json');
ajv.getSchema("https://json-schema.org/draft/2020-12/schema")
const validate_timeseries = ajv.compile(timeseriesSchema);


describe('Timeseries data validation', () => {
  it('validates passing data against the schema', () => {
    const passingData = require('../../data/v0/passing_timeseries_v0.json');
    const isValid = validate_timeseries(passingData);
    expect(validate_timeseries.errors).toBeNull();
    expect(isValid).toBeTruthy(); // Assert that the data passes validation
  });

  it('validates failing data against the schema', () => {
    const failingData = require('../../data/v0/failing_timeseries_v0.json');
    const isValid = validate_timeseries(failingData);
    expect(validate_timeseries.errors).toMatchSnapshot(); // Assert that the data fails validation
  });
})