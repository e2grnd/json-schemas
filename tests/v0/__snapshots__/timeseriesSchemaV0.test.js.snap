// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeseries data validation validates failing data against the schema 1`] = `
[
  {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'dcs-id'",
    "params": {
      "missingProperty": "dcs-id",
    },
    "schemaPath": "#/required",
  },
  {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'units'",
    "params": {
      "missingProperty": "units",
    },
    "schemaPath": "#/required",
  },
  {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'time-bucket-size'",
    "params": {
      "missingProperty": "time-bucket-size",
    },
    "schemaPath": "#/required",
  },
  {
    "instancePath": "",
    "keyword": "required",
    "message": "must have required property 'operation'",
    "params": {
      "missingProperty": "operation",
    },
    "schemaPath": "#/required",
  },
  {
    "instancePath": "/data-csv/storage-key",
    "keyword": "type",
    "message": "must be string",
    "params": {
      "type": "string",
    },
    "schemaPath": "#/properties/data-csv/properties/storage-key/type",
  },
  {
    "instancePath": "/time-bucket-unit",
    "keyword": "enum",
    "message": "must be equal to one of the allowed values",
    "params": {
      "allowedValues": [
        "hour",
        "day",
        "week",
        "month",
        "year",
      ],
    },
    "schemaPath": "#/properties/time-bucket-unit/enum",
  },
]
`;
